api/client/client.go

```diff
@@ -272,21 +272,28 @@ func connect(ctx context.Context, cfg Config) (*Client, error) {
 				})
 			}
 
-			// Connect with dialer provided in creds.
-			if dialer, err := creds.Dialer(cfg); err != nil {
-				if !trace.IsNotImplemented(err) {
-					sendError(trace.Wrap(err, "failed to retrieve dialer from creds of type %T", creds))
-				}
-			} else {
-				syncConnect(ctx, dialerConnect, connectParams{
-					cfg:       cfg,
-					tlsConfig: tlsConfig,
-					dialer:    dialer,
+			addrs := cfg.Addrs
+			if len(addrs) == 0 {
+				// If there's no explicitly specified address, fall back to
+				// an address provided by the credential, if it provides one.
+				credAddrSource, ok := creds.(interface {
+					ProxyWebAddr() (string, error)
```

Comment:
suggestion: It might be worth generalizing this to just `GetDefaultAddrs` and adding it to some exported interface, like `CredentialsWithDefaultAddrs`. This is still intended as an external client which we'd want to standardize and make easy to use for external developers. Someone may for example want to create a credential type that returns an auth addr stored in some file in some file.

Reply:
Sure - I can implement that.

Reply:
17c46eaa7b9d19174c54505576b222849ed0154c